src/pages/Login/LoginPage.tsx

```diff
@@ -199,7 +199,7 @@ export class LoginPage extends React.Component<LoginProps, LoginState> {
         textContent="Service Mesh Observability."
         loginTitle="Log in Kiali"
       >
-        {authenticationConfig.strategy === AuthStrategy.login ? (
+        {authenticationConfig.strategy === AuthStrategy.login || authenticationConfig.strategy === AuthStrategy.ldap ? (
```

Comment:
```diff
diff --git a/src/services/Login.ts b/src/services/Login.ts
index 3695300e..94e3e94c 100644
--- a/src/services/Login.ts
+++ b/src/services/Login.ts
@@ -19,6 +19,7 @@ export interface LoginResult {
 interface LoginStrategy<T extends unknown> {
   prepare: (info: AuthConfig) => Promise<AuthResult>;
   perform: (request: DispatchRequest<T>) => Promise<LoginResult>;
+  requiresLoginForm: () => bool;
 }
 
 interface DispatchRequest<T> {
@@ -45,6 +46,10 @@ class AnonymousLogin implements LoginStrategy<unknown> {
       }
     };
   }
+
+  public requiresLoginForm() {
+    return false;
+  }
 }
 
 interface WebLoginData {
@@ -65,6 +70,10 @@ class WebLogin implements LoginStrategy<WebLoginData> {
       session: session
     };
   }
+
+  public requiresLoginForm() {
+    return true;
+  }
 }
 
 class OpenshiftLogin implements LoginStrategy<unknown> {
@@ -95,6 +104,10 @@ class OpenshiftLogin implements LoginStrategy<unknown> {
       session: session
     };
   }
+
+  public requiresLoginForm() {
+    return false;
+  }
 }
 
 export class LoginDispatcher {
```

Now that we have more types that require different behavior on other places, I think we maybe need to evolve on this front, by extracting the capability instead of special-casing many authorization places, so this is a suggestion, what do you think?

Reply:
@cfcosta I think it is good idea, it will be easier to activate the login form. It will be better than the if condition.